pkg/operator/ceph/csi/secrets.go

```diff
@@ -183,38 +237,199 @@ func CreateCSISecrets(context *clusterd.Context, clusterInfo *client.ClusterInfo
+	rbdProvName, rbdProvKey, rbdProvKeyCount, _, err := createCSIKeyring(context, clusterInfo, cephCluster, k, csiKeyringRBDProvisionerUsername, cephCSIKeyringRBDProvisionerCaps())
 	if err != nil {
 		return errors.Wrap(err, "failed to create csi rbd provisioner ceph keyring")
 	}
 
 	// Create CSI RBD Node Ceph key
-	csiRBDNodeSecretKey, err := createCSIKeyringRBDNode(k)
+	rbdNodeName, rbdNodeKey, rbdNodeKeyCount, _, err := createCSIKeyring(context, clusterInfo, cephCluster, k, csiKeyringRBDNodeUsername, cephCSIKeyringRBDNodeCaps())
 	if err != nil {
 		return errors.Wrap(err, "failed to create csi rbd node ceph keyring")
 	}
 
 	// Create CSI Cephfs provisioner Ceph key
-	csiCephFSProvisionerSecretKey, err := createCSIKeyringCephFSProvisioner(k)
+	cFSProvName, cFSProvKey, cFSProvKeyCount, _, err := createCSIKeyring(context, clusterInfo, cephCluster, k, csiKeyringCephFSProvisionerUsername, cephCSIKeyringCephFSProvisionerCaps())
 	if err != nil {
 		return errors.Wrap(err, "failed to create csi cephfs provisioner ceph keyring")
 	}
 
 	// Create CSI Cephfs node Ceph key
-	csiCephFSNodeSecretKey, err := createCSIKeyringCephFSNode(k)
+	cFSNodeName, cFSNodeKey, cFSNodeKeyCount, shouldRotate, err := createCSIKeyring(context, clusterInfo, cephCluster, k, csiKeyringCephFSNodeUsername, cephCSIKeyringCephFSNodeCaps())
 	if err != nil {
 		return errors.Wrap(err, "failed to create csi cephfs node ceph keyring")
 	}
 
+	csiSecretContent := csiSecretStore{
+		CsiRBDProvisionerSecret:    {Name: rbdProvName, Key: rbdProvKey},
+		CsiRBDNodeSecret:           {Name: rbdNodeName, Key: rbdNodeKey},
+		CsiCephFSProvisionerSecret: {Name: cFSProvName, Key: cFSProvKey},
+		CsiCephFSNodeSecret:        {Name: cFSNodeName, Key: cFSNodeKey},
+	}
+
 	// Create or update Kubernetes CSI secret
-	if err := createOrUpdateCSISecret(clusterInfo, csiRBDProvisionerSecretKey, csiRBDNodeSecretKey, csiCephFSProvisionerSecretKey, csiCephFSNodeSecretKey, k); err != nil {
+	if err := createOrUpdateCSISecret(clusterInfo, csiSecretContent, k); err != nil {
 		return errors.Wrap(err, "failed to create kubernetes csi secret")
 	}
 
+	currentKeyCount := max(rbdProvKeyCount, rbdNodeKeyCount, cFSProvKeyCount, cFSNodeKeyCount)
+	err = updateCephStatusWithCephxStatus(context, clusterInfo, cephCluster, clusterNamespaced, shouldRotate, currentKeyCount)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+// updateCephStatusWithCephxStatus updates the cephCluster status with CSI cephxStatus
+func updateCephStatusWithCephxStatus(context *clusterd.Context, clusterInfo *client.ClusterInfo, cephCluster *cephv1.CephCluster,
+	namespacedName types.NamespacedName, shouldRotate bool, currentKeyCount int,
+) error {
+	logger.Info("updating cephStatus with CSI cephxStatus")
+	cephxStatus := keyring.UpdatedCephxStatus(shouldRotate, cephCluster.Spec.Security.CephX.Daemon, clusterInfo.CephVersion, *cephCluster.Status.Cephx.CSI.CephxStatus)
```

Comment:
Nit:  shouldRotate vs didRotate is only semantic here, but I think it is beneficial for reading the code. The `shouldRotate` returned by `createCSIKeyring()` is returned after rotation has already taken place, which means this indicates whether rotation did happen and not whether rotation is still needed, as "should" might indicate.

```suggestion
func updateCephStatusWithCephxStatus(context *clusterd.Context, clusterInfo *client.ClusterInfo, cephCluster *cephv1.CephCluster,
	namespacedName types.NamespacedName, didRotate bool, currentKeyCount int,
) error {
	logger.Info("updating cephStatus with CSI cephxStatus")
	cephxStatus := keyring.UpdatedCephxStatus(didRotate, cephCluster.Spec.Security.CephX.Daemon, clusterInfo.CephVersion, *cephCluster.Status.Cephx.CSI.CephxStatus)
```